plugins/modules/vmware_guest.py

```diff
@@ -2251,25 +2228,25 @@ def customize_vm(self, vm_obj):
             # https://pubs.vmware.com/vi3/sdk/ReferenceGuide/vim.vm.customization.IPSettings.html
             if 'domain' in network:
                 guest_map.adapter.dnsDomain = network['domain']
-            elif 'domain' in self.params['customization']:
+            elif self.params['customization']['domain'] is not None:
                 guest_map.adapter.dnsDomain = self.params['customization']['domain']
 
             if 'dns_servers' in network:
                 guest_map.adapter.dnsServerList = network['dns_servers']
-            elif 'dns_servers' in self.params['customization']:
+            elif self.params['customization']['dns_servers'] is not None:
                 guest_map.adapter.dnsServerList = self.params['customization']['dns_servers']
 
             adaptermaps.append(guest_map)
 
         # Global DNS settings
         globalip = vim.vm.customization.GlobalIPSettings()
-        if 'dns_servers' in self.params['customization']:
+        if self.params['customization']['dns_servers'] is not None:
             globalip.dnsServerList = self.params['customization']['dns_servers']
 
         # TODO: Maybe list the different domains from the interfaces here by default ?
         dns_suffixes = []
-        if 'dns_suffix' in self.params['customization']:
-            dns_suffix = self.params['customization']['dns_suffix']
+        dns_suffix = self.params['customization']['dns_suffix']
+        if dns_suffix is not None:
```

Comment:
You can remove a level of indentation and just do:
```python
if dns_suffix:
```

Reply:
@goneri Can you please have a look at my last commit? I think it implements what you meant.

Any other suggestions? Or do you think we can merge now?